code-challenges/treeIntersection/tree-intersection.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const HT = require('../../Data-Structures/hashtable/hashtable');
+
+module.exports = (bt1, bt2) => {
+  let result = new HT;
+  let vals1 = bt1.preOrder();
+  let vals2 = bt2.preOrder();
+
+  for (let i of vals1) {
+    if (vals2.includes(i) && !result.contains(i))
```

Comment:
The `vals2.includes(i)` check here is O(n), meaning the overall algoritm is O(n²). Better to use a hashtable (`visited`?) for that lookup. I'd generally allocate a `result` array separate from the hashtable/hashset used for optimization.